decentraland/unity-renderer
# refactor: move audio components to own assembly

## What does this PR change?

This PR moves audio components to their own assembly. Had to remove `GlobalScene` references from audio components to make this work and replaced them with equivalent behaviour.

Actionable for https://github.com/decentraland/unity-renderer/issues/1357

## How to test the changes?

We may want to test portable experiences that use sound and enable they play the sound correctly, although this should be already covered by tests.

Steps for test:
1. Enter the world.
2. Go to Google console (F12).
3. Write `spawnScenePortableExperienceSceneFromUrn("urn:decentraland:off-chain:static-portable-experiences:radio", "main")`
4. You will see the PEX is executed (don't worry about the infinite loading spinner, it is because it is a old PEX).
![image](https://user-images.githubusercontent.com/64659061/152512164-61df4cf7-2879-47c8-b67e-a98ec6ba3727.png)
5. Select any station in the PEX UI.
6. The music is not playing, when it should.

You can also test it with the "Portal" PEX (`spawnScenePortableExperienceSceneFromUrn("urn:decentraland:off-chain:static-portable-experiences:portal-experience", "main")`), when a sound should be playing each time we shoot:
![image](https://user-images.githubusercontent.com/64659061/152512516-585c7de1-9889-473d-b0cf-8a2d3e91feee.png)


## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
